rmw_iceoryx_cpp/src/types/iceoryx_client.hpp

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2022 - 2023 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TYPES__ICEORYX_CLIENT_HPP_
+#define TYPES__ICEORYX_CLIENT_HPP_
+
+#include "iceoryx_posh/popo/untyped_client.hpp"
+
+#include "rmw/rmw.h"
+#include "rmw/types.h"
+
+#include "rmw_iceoryx_cpp/iceoryx_type_info_introspection.hpp"
+
+struct IceoryxClient
+{
+  IceoryxClient(
+    const rosidl_service_type_support_t * type_supports,
+    iox::popo::UntypedClient * const iceoryx_client)
+  : type_supports_(*type_supports),
+    iceoryx_client_(iceoryx_client),
+    is_fixed_size_(rmw_iceoryx_cpp::iceoryx_is_fixed_size(type_supports)),
+    request_size_(rmw_iceoryx_cpp::iceoryx_get_request_size(type_supports))
+  {}
+
+  rosidl_service_type_support_t type_supports_;
+  iox::popo::UntypedClient * const iceoryx_client_;
+  bool is_fixed_size_;
+  size_t request_size_;
+  /// @todo How can the full type be aquired via rosidl to do an 'alignof()'?
```

Comment:
@milancpp Can you answer the above question?

Reply:
@sumanth-nirmal might know

Reply:
@mossmaurice Since we use dynamic introspection type support, I think we should iterate through the members of the type and determine the alignment.

The introspection type support has a pointer to the members https://github.com/ros2/rosidl/blob/rolling/rosidl_typesupport_introspection_cpp/include/rosidl_typesupport_introspection_cpp/message_introspection.hpp#L100 and offset of each field https://github.com/ros2/rosidl/blob/rolling/rosidl_typesupport_introspection_cpp/include/rosidl_typesupport_introspection_cpp/message_introspection.hpp#L52

An example for interating through the members can be seen here https://github.com/ros2/rmw_cyclonedds/blob/87f0272476da5c00e35ff6c6c5c9fdba421906dd/rmw_cyclonedds_cpp/src/TypeSupport_impl.hpp#L264-L347

Reply:
@sumanth-nirmal Ok, thanks! In `rmw_iceoryx` we also iterate for [serializing](https://github.com/ros2/rmw_iceoryx/blob/master/rmw_iceoryx_cpp/src/internal/iceoryx_serialize_typesupport_cpp.hpp#L152) and [deserializing](https://github.com/ros2/rmw_iceoryx/blob/master/rmw_iceoryx_cpp/src/internal/iceoryx_deserialize_typesupport_cpp.hpp#L209). I'll think I'll probably write a custom method which calculates the alignment.

Reply:
@budrus After talking to @elBoberido we concluded that `iox::CHUNK_DEFAULT_USER_PAYLOAD_ALIGNMENT` should be sufficient. I opened the issue to provide a default arg like in pub/sub: https://github.com/eclipse-iceoryx/iceoryx/issues/1845